docs/blog/pydantic-v2.md

```diff
@@ -0,0 +1,945 @@
+    see [pydantic-core#155](https://github.com/samuelcolvin/pydantic-core/issues/155)
+15. We'll now follow [semvar](https://semver.org/) properly and avoid breaking changes between minor versions,
+    as a result, major versions will become more common
+
+## Removed Features & Limitations :frowning:
+
+The emoji here is just for variation, I'm not frowning about any of this, these changes are either good IMHO
+(will make pydantic cleaner, easier to learn and easier to maintain) or irrelevant to 99.9+% of users.
+
+1. `__root__` custom root models are no longer necessary since validation on any supported data type is allowed
+   without a model
+2. `.parse_file()` and `.parse_raw()`, partially replaced with `.model_validate_json()`, 
+   see [model methods](#model-namespace-cleanup)
+3. `.schema_json()` & `.copy()`, see [model methods](#model-namespace-cleanup)
+4. `TypeError` are no longer considered as validation errors, but rather as internal errors, this is to better
+   catch errors in argument names in function validators.
+5. Subclasses of builtin types like `str`, `bytes` and `int` are coerced to their parent builtin type,
+   this is a limitation of how pydantic-core converts these types to Rust types during validation, if you have a
+   specific need to keep the type, you can use wrap validators or custom type validation as described above
+6. integers are represented in rust code as `i64`, meaning if you want to use ints where `abs(v) > 2^63 − 1`
+   (9,223,372,036,854,775,807), you'll need to use a [wrap validator](#validator-function-improvements) and your own logic
+7. [Settings Management](https://pydantic-docs.helpmanual.io/usage/settings/) ??? - I definitely don't want to
+   remove the functionality, but it's something of a historical curiosity that it lives within pydantic,
+   perhaps it should move to a separate package, perhaps installable alongside pydantic with
+   `pip install pydantic[settings]`?
+8. The following `Config` properties will be removed:
+   * `fields` - it's very old (it pre-dates `Field`), can be removed
+   * `allow_mutation` will be removed, instead `frozen` will be used
+   * `error_msg_templates`, it's not properly documented anyway, error messages can be customized with external logic if required
+   * `getter_dict` - pydantic-core has hardcoded `from_attributes` logic
+   * `json_loads` - again this is hard coded in pydantic-core
+   * `json_dumps` - possibly
+   * `json_encoders` - see the export "mode" discussion [above](#improvements-to-dumpingserializationexport)
+   * `underscore_attrs_are_private` we should just choose a sensible default
+   * `smart_union` - all unions are now "smart"
+9. `dict(model)` functionality should be removed, there's a much clearer distinction now that in 2017 when I 
+   implemented this between a model and a dict
+
+## Features Remaining :neutral_face:
+
+The following features will remain (mostly) changed:
+
+* Generics
```

Comment:
I think this should change and would be a very nice point in the "other improvements" section. Now that we stop using cython, we should be able to have 
```py
class M(Basemodel, Generic [T])...
```
instead of the `GenericModel`